amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutConfig.java

```diff
@@ -82,26 +82,26 @@
 
     @Override
     public RetrievalFactory retrievalFactory() {
-        return new FanOutRetrievalFactory(kinesisClient, getOrCreateConsumerArn());
+        return new FanOutRetrievalFactory(kinesisClient, streamName, this::getOrCreateConsumerArn);
     }
 
-    private String getOrCreateConsumerArn() {
+    private String getOrCreateConsumerArn(String streamName) {
```

Comment:
is stream name sufficient here? stream creation timestamp sounds like the other critical piece of information for consumer creation.

Reply:
Hmmm, let's sync up on this offline.